IPython/core/interactiveshell.py

```diff
@@ -283,49 +283,55 @@ def _exiter_default(self):
     input_transformer_manager = Instance('IPython.core.inputsplitter.IPythonInputSplitter',
                                          (), {'line_input_checker': False})
     
-    logstart = CBool(False, config=True, help=
+    logstart = CBool(False).tag(config=True, help=
         """
         Start logging to the default log file in overwrite mode.
         Use `logappend` to specify a log file to **append** logs to.
         """
     )
-    logfile = Unicode('', config=True, help=
+    logfile = Unicode('').tag(config=True, help=
         """
         The name of the logfile to use.
         """
     )
-    logappend = Unicode('', config=True, help=
+    logappend = Unicode('').tag(config=True, help=
         """
         Start logging to the given file in append mode.
         Use `logfile` to specify a log file to **overwrite** logs to.
         """
     )
-    object_info_string_level = Enum((0,1,2), default_value=0,
+    object_info_string_level = Enum((0,1,2), default_value=0).tag(
                                     config=True)
-    pdb = CBool(False, config=True, help=
+    pdb = CBool(False).tag(config=True, help=
         """
         Automatically call the pdb debugger after every exception.
         """
     )
-    multiline_history = CBool(sys.platform != 'win32', config=True,
+    multiline_history = CBool(sys.platform != 'win32').tag(config=True,
         help="Save multi-line entries as one entry in readline history"
     )
-    display_page = Bool(False, config=True,
+    display_page = Bool(False).tag(config=True,
         help="""If True, anything that would be passed to the pager
         will be displayed as regular output instead."""
     )
 
     # deprecated prompt traits:
     
-    prompt_in1 = Unicode('In [\\#]: ', config=True,
+    prompt_in1 = Unicode('In [\\#]: ').tag(config=True,
         help="Deprecated, will be removed in IPython 5.0, use PromptManager.in_template")
-    prompt_in2 = Unicode('   .\\D.: ', config=True,
+    prompt_in2 = Unicode('   .\\D.: ').tag(config=True,
         help="Deprecated, will be removed in IPython 5.0, use PromptManager.in2_template")
-    prompt_out = Unicode('Out[\\#]: ', config=True,
+    prompt_out = Unicode('Out[\\#]: ').tag(config=True,
         help="Deprecated, will be removed in IPython 5.0, use PromptManager.out_template")
-    prompts_pad_left = CBool(True, config=True,
+    prompts_pad_left = CBool(True).tag(config=True,
         help="Deprecated, will be removed in IPython 5.0, use PromptManager.justify")
-    
+
+   
+    @observe('prompt_trait',
```

Comment:
I don't think there's a `prompt_trait` trait, I think that's just what I called the method.